roles/pulp_database/tasks/install_postgres.yml

```diff
@@ -1,29 +1,111 @@
 ---
 - block:
+    - name: Check if PostgreSQL < 10 is initialized.
+      stat:
+        path: "{{ postgresql_previous_data_dir }}/PG_VERSION"
+      register: previous_pgdata_dir_version
+
+    - name: Get PG_VERSION
+      command: "cat {{ postgresql_previous_data_dir }}/PG_VERSION"
+      register: pg_version
+      when: previous_pgdata_dir_version.stat.exists
+      tags:
+        - molecule-idempotence-notest
+
+    - name: Stop PostgreSQL < 10 service
+      systemd:
+        name: "{{ postgresql_previous_daemon }}"
+        state: stopped
+        enabled: no
+      register: pgservice_result
+      failed_when: "pgservice_result is failed and 'Service is in unknown state' not in pgservice_result.msg"
+      when:
+        - previous_pgdata_dir_version.stat.exists
+        - pg_version is defined
+        - pg_version.changed
+        - pg_version.stdout is version('10', '<')
+
     - name: Install SCL package
       package:
-        name: 'centos-release-scl'
+        name:
+          - 'scl-utils'
+          - 'centos-release-scl-rh'
+          - 'centos-release-scl'
         state: present
-      when:
-        - ansible_facts.distribution == 'CentOS'
-        - ansible_facts.distribution_major_version|int == 7
 
     - name: Install PostgreSQL SCL
-      package:
-        name: 'rh-postgresql96'
+      yum:
+        name: "{{ postgresql_packages }}"
         state: present
-      when:
-        - ansible_facts.distribution == 'CentOS'
-        - ansible_facts.distribution_major_version|int == 7
+        exclude: python-unversioned-command
 
     - name: Enable PostgreSQL SCL
       template:
         src: templates/postgresql_scl_profile.j2
         dest: "{{ postgresql_profile_path }}"
+
+    - name: Create SCL symlinks
+      file:
+        src: "{{ item }}"
+        dest: "{{ item.replace('/opt/rh/rh-postgresql10/root', '') }}"
+        state: link
+      loop:
+        - "/opt/rh/rh-postgresql10/root/usr/lib64/libpq.so.rh-postgresql10-5"
+        - "/opt/rh/rh-postgresql10/root/usr/bin/pg_config"
+      register: symlink_result
+      failed_when: "symlink_result is failed and 'refusing to convert from file to symlink' not in symlink_result.msg"
+      check_mode: false
```

Comment:
I get what you are doing, but is this the only way?

Can't we not set the environment LD_LIBRARY_PATH for our tasks?

(Last  I tried, we cannot set it for geerlingguy's role tasks that run as sudo.)

Reply:
I didn't think about that, but the problem is:
we were getting errors on pip install pulp plugins, so I would need to carry this through couple of roles only due to CentOS 7

Reply:
Can't we set LD_LIBRARY_PATH for the pip install plugins tasks?

Reply:
I can try

Reply:
I removed the symlink and it broke on geerling's role:
```
TASK [geerlingguy.postgresql : Ensure PostgreSQL users are configured correctly.] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: error:060800A3:digital envelope routines:EVP_DigestInit_ex:disabled for fips
failed: [pulp3-source-centos7-fips] (item={'name': 'pulp', 'password': 'pulp', 'role_attr_flags': 'CREATEDB'}) => {"ansible_loop_var": "item", "changed": false, "item": {"name": "pulp", "password": "pulp", "role_attr_flags": "CREATEDB"}, "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 102, in <module>\n  File \"<stdin>\", line 94, in _ansiballz_main\n  File \"<stdin>\", line 40, in invoke_module\n  File \"/usr/lib64/python2.7/runpy.py\", line 176, in run_module\n    fname, loader, pkg_name)\n  File \"/usr/lib64/python2.7/runpy.py\", line 82, in _run_module_code\n    mod_name, mod_fname, mod_loader, pkg_name)\n  File \"/usr/lib64/python2.7/runpy.py\", line 72, in _run_code\n    exec code in run_globals\n  File \"/tmp/ansible_postgresql_user_payload_FFi9DT/ansible_postgresql_user_payload.zip/ansible_collections/community/general/plugins/modules/postgresql_user.py\", line 993, in <module>\n  File \"/tmp/ansible_postgresql_user_payload_FFi9DT/ansible_postgresql_user_payload.zip/ansible_collections/community/general/plugins/modules/postgresql_user.py\", line 932, in main\n  File \"/tmp/ansible_postgresql_user_payload_FFi9DT/ansible_postgresql_user_payload.zip/ansible_collections/community/general/plugins/modules/postgresql_user.py\", line 453, in user_alter\n  File \"/tmp/ansible_postgresql_user_payload_FFi9DT/ansible_postgresql_user_payload.zip/ansible_collections/community/general/plugins/modules/postgresql_user.py\", line 419, in user_should_we_change_password\nValueError: error:060800A3:digital envelope routines:EVP_DigestInit_ex:disabled for fips\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

Reply:
here is the version without symlink:
https://github.com/pulp/pulp_installer/pull/538

Reply:
I think the symlink version is less invasive